driver-core/src/main/com/mongodb/client/model/WindowedComputations.java

```diff
@@ -0,0 +1,765 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model;
+
+import com.mongodb.client.model.Windows.Bound;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentWriter;
+import org.bson.BsonType;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.client.model.MongoTimeUnit.MONTH;
+import static com.mongodb.client.model.MongoTimeUnit.YEAR;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Builders for {@linkplain WindowedComputation windowed computations} used in the
+ * {@link Aggregates#setWindowFields(Object, Bson, List) $setWindowFields} pipeline stage
+ * of an aggregation pipeline. Each windowed computation is a triple:
+ * <ul>
+ *     <li>A window function. Some functions require documents in a window to be sorted
+ *     (see {@code sortBy} in {@link Aggregates#setWindowFields(Object, Bson, List)}).</li>
+ *     <li>An optional {@linkplain Window window}, a.k.a. frame.
+ *     Specifying {@code null} window is equivalent to specifying [{@link Bound#UNBOUNDED}, {@link Bound#UNBOUNDED}].</li>
+ *     <li>A path to an output field to be computed by the window function over the window.</li>
+ * </ul>
+ * A windowed computation is similar to an {@linkplain Accumulators accumulator} but does not result in folding documents constituting
+ * the window into a single document.
+ *
+ * @mongodb.driver.manual meta/aggregation-quick-reference/#field-paths Field paths
+ * @since 4.3
+ */
+public final class WindowedComputations {
+    /**
+     * Builds a computation of the sum of the evaluation results of the {@code expression} over the {@code window}.
+     *
+     * @param path The output field path.
+     * @param expression The expression.
+     * @param window The window.
+     * @param <TExpression> The expression type.
+     * @return The constructed {@link WindowedComputation}.
+     * @mongodb.driver.dochub core/window-functions-sum $sum
+     * @mongodb.server.release 5.0
+     */
+    public static <TExpression> WindowedComputation sum(final String path, final TExpression expression, @Nullable final Window window) {
```

Comment:
In the new documentation example for `$setWindowFields` I used `WindowedComputations.sum` without using static import.

It may be worth pointing out that having the names of the `WindowedComputations` methods clashing with those in the `Accumulators` class does not break compilation for existing code, as it was the case in https://github.com/mongodb/mongo-java-driver/pull/738.

Reply:
It's only a problem if the method signature is identical.